### PR TITLE
docs: strikethrough deprecated methods on api index

### PR DESCRIPTION
**Pull request by matthewmayer**
Strikes through deprecated methods on API index page so its easier to see the non-deprecated methods

Example:
Screenshot 2023-04-16 at 20 12 59

### CODE REVIEW

File: docs/api/ApiIndex.vue [hunk 1]
```diff
@@ -165,6 +169,9 @@ h3 {
   color: var(--vp-c-text-code);
   transition: color 0.5s;
 }
+.api-groups ul a.deprecated {
```
Comment (ST-DDT) [inline docs/api/ApiIndex.vue:173]:
```suggestion

.api-groups ul a.deprecated {
```

**Review by ST-DDT — COMMENTED** [review]
I somewhat like it, but I'm not sure whether the user easily understands that the methods there are deprecated/what the strike-through should mean.

Also I'm not sure whether this adds any kind of benefit. If you notice a method doing what you search for then you click on it. If it happens to be deprecated, you will read the deprecation hint pointing to the right method.

**Comment by matthewmayer** [comment]
The idea is if you are skimming looking for the right method you can focus on the ones that are non-deprecated.

**Comment by ST-DDT** [comment]
**Team Decision**

- We want this.

**Review by xDivisionByZerox — DISMISSED** [review]

File: docs/api/ApiIndex.vue [hunk 2]
```diff
@@ -101,7 +101,11 @@ onUnmounted(() => window.removeEventListener('keydown', apiSearchFocusHandler));
           </h3>
           <ul>
             <li v-for="h of item.headers" :key="h.anchor">
-              <a :href="item.link + '#' + slugify(h.anchor)">{{ h.text }}</a>
+              <a
+                :href="item.link + '#' + slugify(h.anchor)"
+                :class="h.deprecated ? 'deprecated' : ''"
```
Comment (ST-DDT) [inline docs/api/ApiIndex.vue]: Should we simplify this to:

```suggestion
                :class="{ deprecated: h.deprecated }"
```

?

**Review by ST-DDT — DISMISSED** [review]
Looks good to me.

File: docs/api/ApiIndex.vue (hunk 2 above)
Comment (Shinigami92) [reply docs/api/ApiIndex.vue]: yes, we should

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**